src/cmdstan/io/json/json_data.hpp

```diff
@@ -119,6 +120,43 @@ class json_data : public stan::io::var_context {
     return empty_vec_r_;
   }
 
+  /**
+   * Read out a vector of complex numbers from the specifed name.
+   * @param name Name of Variable.
+   */
+  std::vector<std::complex<double>> vals_c(const std::string &name) const {
+    if (contains_r_only(name)) {
+      auto &&vec_r = (vars_r_.find(name)->second);
+      auto &&val_r = vec_r.first;
+      auto &&dim_r = vec_r.second;
+      std::vector<std::complex<double>> vec_c(val_r.size() / 2);
+      int offset = 1;
+      for (int i = 0; i < dim_r.size() - 1; ++i) {
```

Comment:
[question]
why is this less than size - 1?  Don't you want to take the product of all of the elements? '

[optional]
You can convert the initialization and loop into a one-liner functional form using `std::accumulate`.

Reply:
Not sure why we're doing size - 1. Steve wrote this code and might have the answer for this.

Reply:
Coming back to it, I now see that it's just multiplying out first N - 1 dimension sizes so it can index into the last dimension.  So this is fine.